src/Psalm/Internal/Cache.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Psalm\Internal;
+
+use Psalm\Config;
+use Psalm\Internal\Provider\Providers;
+
+use function file_exists;
+use function file_put_contents;
+use function gzdeflate;
+use function gzinflate;
+use function igbinary_serialize;
+use function igbinary_unserialize;
+use function serialize;
+use function unlink;
+use function unserialize;
+
+use const LOCK_EX;
+
+/**
+ * @internal
+ */
+class Cache
+{
+    private Config $config;
+
+    public bool $use_igbinary;
+
+    public function __construct(Config $config)
+    {
+        $this->config = $config;
+        $this->use_igbinary = $config->use_igbinary;
+    }
+
+    /**
+     * @return array|object|string|null
+     */
+    public function getItem(string $path)
+    {
+        if (!file_exists($path)) {
```

Comment:
`is_file` is faster than `file_exists`, PHP does some opcache optimizations with former

Reply:
I can't see any difference one way or another.